views/country_download.erb

```diff
@@ -132,68 +132,42 @@ var max_visible_terms = 6;
-  // class = "hidden-term" and then hiding them,
-  // display JS control for toggling them
-
-$(".country__legislature").each(function(){
-  var house_slug = $(this).data('house');
-  var $term_list_items = $('#terms-' + house_slug + ' li');
-  if ($term_list_items.size() > max_visible_terms) {
-    $term_list_items.addClass("hidden-term");
-    var wanted_house_index = 0;
-    // we need to hide some of these terms:
-    // find the one we want:
-    var wanted_id = "term-" + house_slug + '-' + wanted_term;
-    // want to use jQuery's .index() but it's not behaving as I expected
-    if (wanted_house && wanted_term &&
-      wanted_house == house_slug
-    ) {
-      $term_list_items.each(function(i){
-        if ($(this).attr('id') == wanted_id) {
-          wanted_house_index = i;
-          $(this).css("background-color", "#fffadd"); // highlight (FIXME?)
-          return false; // break
-        }
-      });
-    }
-    // want a slice around wanted_house_index of reduced_visible_terms items
-    var max_upper_bound = $term_list_items.length - 1;
-    var lower_bound = wanted_house_index - Math.floor(reduced_visible_terms/2);
-    var upper_bound = wanted_house_index + Math.floor(reduced_visible_terms/2);
-    var delta = upper_bound - max_upper_bound;
-    if (delta > 0) {
-      upper_bound = max_upper_bound;
-      lower_bound = lower_bound - delta;
+    if (wanted_house == house_slug && wanted_term) {
+      $wanted_term_element = $('#' + "term-" + house_slug + '-' + wanted_term);
+      $wanted_term_element.addClass("download-terms__highlighted");
     }
-    if (lower_bound < 0) {
-      upper_bound = Math.min(upper_bound + Math.abs(lower_bound), max_upper_bound);
-      lower_bound = 0;
-    }
-    // would do this with jQuery's slice but it's not behaving as I expected
-    $term_list_items.each(function(i){
-      if (i >= lower_bound && i <= upper_bound) {
-        $(this).removeClass("hidden-term");
-      }
-    });
-    // add button to toggle view
-    $term_list_items.first().parent().before('<a href="#" class="button toggle-term-display" data-house="'
-    + house_slug + '">Show all ' + $term_list_items.length + ' terms</a>');
-  }
-});
-$('.hidden-term').hide();
-$('.button.toggle-term-display').on("click", function(e){
-  e.preventDefault();
-  var house = $(e.target).data("house");
-  $('#terms-' + house + " li.hidden-term").slideToggle();
-})
+
+    var $button_for_reveal = $(
+       '<a href="#" class="button toggle-term-display" data-house="'
+       + house_slug + '">Show all ' + $term_list_items.length + ' terms</a>');
+
+    collapse_displayed_items(
+      $term_list_items,
+      $wanted_term_element, // note: may be undefined
+      'hidden-term',
+      max_visible_terms,
+      reduced_visible_terms,
+      $button_for_reveal
+    );
+  });
+
+  $('.hidden-term').hide();
+  $('.button.toggle-term-display').on("click", function(e){
+    e.preventDefault();
+    var house = $(e.target).data("house");
```

Comment:
I think you want `$(this)` here, instead of `$(e.target)`.

In most situations, they’re equivalent. But imagine the situation where we wrap a `<span>` around the text content in the `toggle-term-display` element – clicking the `span` will trigger the `click` listener on the parent button, but `$(e.target)` will still be set to the `span`, which you don’t want. `$(this)` is what you want, as it will always be whatever element the event listener was bound to (`.toggle-term-display` in this case).